### PR TITLE
8993 nane

### PR DESCRIPTION
**Pull request by shumway-tm**

### CODE REVIEW

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (codyseibert) [inline web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: I'd refactor and put any logic similar to this inside a computed.  We try to avoid having logic in our react components so that we can more easily test them in the computeds code.

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (codyseibert) [inline web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: same as comment above

File: web-client/src/views/CaseDetailEdit/ReviewSavedPetition.jsx
Comment (codyseibert) [inline web-client/src/views/CaseDetailEdit/ReviewSavedPetition.jsx]: this logic should be moved to a computed

File: web-client/src/ustc-ui/Tabs/Tabs.jsx
Comment (codyseibert) [inline web-client/src/ustc-ui/Tabs/Tabs.jsx]: I'm curious why draftCount is a string with a numeric value.  Shouldn't this just be a javascript number?

File: web-api/src/applicationContext.js
Comment (codyseibert) [inline web-api/src/applicationContext.js]: remove the `.js` from the require path

File: web-client/src/ustc-ui/Tabs/Tabs.jsx
Comment (codyseibert) [inline web-client/src/ustc-ui/Tabs/Tabs.jsx:93]: This comment is optional, but I wanted to point it out: 

It feels strange that this generic Tabs component has logic related to notifications and unread messages.  I know we're following the example on line 87, but I think this might warrant a devex refactor to have any additional icons passed in as a prop instead of having logic specific to notifications / draftCounts.

**Review by codyseibert — APPROVED** [review]

File: web-client/src/ustc-ui/Tabs/Tabs.jsx
Comment (shumway-tm) [reply web-client/src/ustc-ui/Tabs/Tabs.jsx:93]: We will create a DEVEX task to address this feedback!

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (rachelschneiderman) [inline web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: Is there a reason to pass in this helper instead of retrieving it off state?

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (ujahio) [reply web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: Refactor completed.

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (ujahio) [reply web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: Refactor completed.

File: web-client/src/views/CaseDetailEdit/ReviewSavedPetition.jsx
Comment (ujahio) [reply web-client/src/views/CaseDetailEdit/ReviewSavedPetition.jsx]: Refactor completed.

File: web-client/src/ustc-ui/Tabs/Tabs.jsx
Comment (ujahio) [reply web-client/src/ustc-ui/Tabs/Tabs.jsx]: We will create a DEVEX task to address this feedback!

File: web-api/src/applicationContext.js
Comment (ujahio) [reply web-api/src/applicationContext.js]: linting issue fixed.

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (ujahio) [reply web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: No particular reason at all. Just following the previous pattern. We can make that suggested adjustment. No worries

File: shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js
Comment (TomElliottFlexion) [inline shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js]: Take it or leave it, but it might be clearer to add the conditional before calling this function

File: web-client/src/ustc-ui/Tabs/Tabs.jsx
Comment (TomElliottFlexion) [inline web-client/src/ustc-ui/Tabs/Tabs.jsx]: Same question here, super curious

**Review by TomElliottFlexion — CHANGES_REQUESTED** [review]

**Review by TomElliottFlexion — APPROVED** [review]

File: shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.test.js
Comment (TomElliottFlexion) [inline shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.test.js:733]: Nit: It's probably time to split this into two test files

**Comment by halprin** [comment]
Remember, don't merge this yet even if we have all the code review approvals.

File: cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js
Comment (rosiehsmith) [inline cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js]: Is this log needed?

File: web-client/src/ustc-ui/Tabs/Tabs.jsx
Comment (rosiehsmith) [reply web-client/src/ustc-ui/Tabs/Tabs.jsx]: Just a suggestion, maybe instead of DEVEX this could be done in the DoD (definition of done) for this story?

File: shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js
Comment (TomElliottFlexion) [reply shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js]: Despite what Github says, this is a suggestion, not a change request

File: shared/src/business/entities/EntityConstants.js
Comment (rosiehsmith) [inline shared/src/business/entities/EntityConstants.js]: Wondering if this could follow the pattern that is being using with SYSTEM_GENERATED_DOCUMENT_TYPES.standingPretrialOrder for example?

Comment (ujahio) [reply shared/src/business/entities/EntityConstants.js]: Refactor complete.

File: cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js
Comment (ujahio) [reply cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js]: Log removed. Good catch

File: shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.test.js
Comment (ujahio) [reply shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.test.js:733]: I agree.

File: shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js
Comment (ujahio) [reply shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js]: Thank you. Looking into how to make those changes.

File: web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx
Comment (ujahio) [reply web-client/src/views/StartCaseInternal/OrdersNeededSummary.jsx]: Refactor completed.

File: web-client/integration-tests/journey/petitionsClerkCreatesNewCaseFromPaper.js
Comment (rachelschneiderman) [inline web-client/integration-tests/journey/petitionsClerkCreatesNewCaseFromPaper.js:257]: Could this be updated to expect some of the new properties that were added to the helper?

File: web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx
Comment (rachelschneiderman) [inline web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx]: Nit: I believe this button is to remove the signature off the docket entry, not remove the docket entry entirely?

File: web-client/src/views/DocketRecord/DraftDocumentViewer.jsx
Comment (rachelschneiderman) [inline web-client/src/views/DocketRecord/DraftDocumentViewer.jsx]: Nit: can you add another '-' between id and the interpolated value please?

File: web-client/src/presenter/computeds/caseDetailSubnavHelper.js
Comment (rachelschneiderman) [inline web-client/src/presenter/computeds/caseDetailSubnavHelper.js]: Nit: Could this variable be renamed to be a bit more specific, only asking because "work items" is another type of item in DAWSON and I would not want it to get confused!

File: cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js
Comment (rachelschneiderman) [inline cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js:1]: Out of curiosity, what was the reason for this being a cypress test over an integration test?

**Review by rachelschneiderman — COMMENTED** [review]
👏 👏 👏 Thank you both for your attention to detail in extracting out small pieces of work into testable functions (ex - getDraftWorkItems). Great job!

File: web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx
Comment (ujahio) [reply web-client/src/views/DocketRecord/DraftDocumentViewerDocument.jsx]: change implemented.

File: web-client/src/presenter/computeds/caseDetailSubnavHelper.js
Comment (ujahio) [reply web-client/src/presenter/computeds/caseDetailSubnavHelper.js]: change implemented.

File: cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js
Comment (ujahio) [reply cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js:1]: The primary reason for making this test a cypress integration was because we were introducing new UI functionality (number of draft items) and wanted to write tests around the new UI implementation/flow.

cc: @halprin

Comment (halprin) [reply cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js:1]: Hmmm, yes agreed, @Absolutestunna.  @rachelschneiderman, I could also be naive about when it is best to have a cypress test versus an integration test.  Is there a good rule of thumb?

Comment (TomElliottFlexion) [reply cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js:1]: Good question

Comment (rachelschneiderman) [reply cypress-integration/integration/petition-clerk-verifies-auto-generated-nane.spec.js:1]: Ok I totally get that! Thanks!

I think in general, I try to avoid adding Cypress tests as they are more expensive to write, run, and less reliable and more brittle than integration tests. Since most of the work done was abstracted into computeds and the `serveCaseToIrsInteractor` I think it could be tested with a jest integration test up until the point where it gets rendered on the page (which in the past we typically have not tested unless we are writing a smoketest/e2e test).

**Review by rachelschneiderman — APPROVED** [review]

**Review by rosiehsmith — APPROVED** [review]

**Pull request merged**
